daemon/api.go

```diff
@@ -1112,12 +1112,19 @@ func trySnap(c *Command, r *http.Request, user *auth.UserState, trydir string, f
 	if !filepath.IsAbs(trydir) {
 		return BadRequest("cannot try %q: need an absolute path", trydir)
 	}
-	if !osutil.IsDirectory(trydir) {
-		return BadRequest("cannot try %q: not a snap directory", trydir)
-	}
 
 	// the developer asked us to do this with a trusted snap dir
 	info, err := unsafeReadSnapInfo(trydir)
+	if _, ok := err.(snap.NoSnapDirError); ok {
```

Comment:
This feels like a missing snap directory, when we supposedly were handed a snap directory and seem to be unhappy about it. How about.. hmmm.. `snap.NotSnapError`?  This way we can use it every time we're sure the content we've been provided is not even a snap (dir or file). We shouldn't use this when we are handed a snap, but its metadata is invalid (`snap.InvalidSnapError` would be better for this).